ballerina/tests/test.bal

```diff
@@ -0,0 +1,176 @@
+        userAccountControl: "544"
+    };
+    LDAPResponse val = check ldapClient->add("CN=User,OU=People,DC=ad,DC=windows", user);
+    test:assertEquals(val.resultStatus, SUCCESS);
+}
+
+@test:Config {
+    dependsOn: [testAddUser]
+}
+public function testAddUserWithManager() returns error? {
+    record {} user = {
+        "objectClass": "user",
+        "sn": "New User",
+        "cn": "New User",
+        "givenName": "New User",
+        "displayName": "New User",
+        "userPrincipalName": "newuser@ad.windows",
+        "userAccountControl": "544",
+        "manager": "CN=User,OU=People,DC=ad,DC=windows"
+    };
+    LDAPResponse addResult = check ldapClient->add("CN=New User,OU=People,DC=ad,DC=windows", user);
+    test:assertEquals(addResult.resultStatus, SUCCESS);
+}
+
+@test:Config {
+    dependsOn: [testGetUser]
+}
+public function testDeleteUserHavingManager() returns error? {
+    LDAPResponse val = check ldapClient->delete("CN=New User,OU=People,DC=ad,DC=windows");
+    test:assertEquals(val.resultStatus, SUCCESS);
+}
+
+@test:Config {
+    dependsOn: [testDeleteUserHavingManager]
+}
+public function testDeleteUser() returns error? {
+    LDAPResponse val = check ldapClient->delete("CN=User,OU=People,DC=ad,DC=windows");
+    test:assertEquals(val.resultStatus, SUCCESS);
+}
+
+@test:Config {
+    dependsOn: [testAddUserWithManager]
+}
+public function testAddAlreadyExistingUser() returns error? {
+    UserConfig user = {
+        objectClass: "user",
+        sn: "New User",
+        cn: "New User",
+        givenName: "New User",
+        displayName: "New User",
+        userPrincipalName: "newuser@ad.windows"
+    };
+    LDAPResponse|Error val = ldapClient->add("CN=New User,OU=People,DC=ad,DC=windows", user);
+    test:assertTrue(val is Error);
+    if val is Error {
+        ErrorDetails errorDetails = val.detail();
+        test:assertEquals(errorDetails.resultStatus, "ENTRY ALREADY EXISTS");
+    }
+}
+
+@test:Config {
+    dependsOn: [testAddAlreadyExistingUser]
+}
+public function testUpdateUser() returns error? {
+    record {} user = {
+        "sn": "User",
+        "givenName": "Updated User",
+        "displayName": "Updated User",
+        "manager": "CN=New User,OU=People,DC=ad,DC=windows"
+    };
+    LDAPResponse val = check ldapClient->modify(userDN, user);
+    test:assertEquals(val.resultStatus, SUCCESS);
+}
+
+@test:Config {
+    dependsOn: [testUpdateUserWithNullValues]
+}
+public function testGetUser() returns error? {
+    UserConfig value = check ldapClient->getEntry(userDN);
+    test:assertEquals(value?.givenName, "Updated User");
+}
+
+@test:Config {}
+public function testInvalidClient() returns error? {
+    Client|Error ldapClient = new ({
+        hostName: "111.111.11.111",
+        port: port,
+        domainName: domainName,
+        password: password
+    });
+    test:assertTrue(ldapClient is Error);
+}
+
+@test:Config {}
+public function testInvalidDomainInClient() returns error? {
```

Comment:
```suggestion
public function testInvalidDomainInClient() {
```

Reply:
Fixed in [cdde708](https://github.com/ballerina-platform/module-ballerina-ldap/pull/45/commits/cdde70852f32464ef384105c5366885fa9486d98)